pkg/es/config/config.go

```diff
@@ -288,6 +292,11 @@ func (c *Configuration) getConfigOptions(logger *zap.Logger) ([]elastic.ClientOp
 		// we don' have a valid token to do the check ad if we don't disable the check the service that
 		// uses this won't start.
 		elastic.SetHealthcheck(!c.AllowTokenFromContext)}
+	if c.SnifferTlsEnabled {
+		options = append(options, elastic.SetScheme("https"))
```

Comment:
Could we derive the scheme from `es.server-urls`? That way we don't have to add new flag.

Reply:
Using the scheme from es-server-urls would work for us because both ingress and the elasticsearch nodes are using https.  So, I'm happy to make this change.  However, strictly speaking the two things are independent of each other.  The point of using sniffing is that we can use ingress/loadbalancer for the initial node discovery then connect to the actual elasticsearch nodes for sending traffic.  Are you concerned about somebody that has an https enabled ingress/loadbalancer but uses http for the elasticsearch nodes?

Reply:
I thought that people using TLS would normally protect all endpoints. If that is not the case we can merge your patch that adds an additional flag to control TLS just for sniffing. 

The PR looks good I have suggested two edits. Once that is done we can merge it.